mbtiles/src/summary.rs

```diff
@@ -0,0 +1,370 @@
+use crate::{MbtResult, MbtType, Mbtiles};
+use serde::Serialize;
+use size_format::SizeFormatterBinary;
+use sqlx::{query, SqliteExecutor};
+use std::{
+    fmt::{Display, Formatter},
+    path::PathBuf,
+    str::FromStr,
+};
+use tilejson::Bounds;
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct ZoomInfo {
+    pub zoom: u8,
+    pub tile_count: u64,
+    pub min_tile_size: u64,
+    pub max_tile_size: u64,
+    pub avg_tile_size: f64,
+    pub bbox: Bounds,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct Summary {
+    pub file_path: String,
+    pub file_size: Option<u64>,
+    pub mbt_type: MbtType,
+    pub page_size: u64,
+    pub page_count: u64,
+    pub tile_count: u64,
+    pub min_tile_size: Option<u64>,
+    pub max_tile_size: Option<u64>,
+    pub avg_tile_size: f64,
+    pub bbox: Option<Bounds>,
+    pub min_zoom: Option<u8>,
+    pub max_zoom: Option<u8>,
+    pub zoom_info: Vec<ZoomInfo>,
+}
+
+impl Display for Summary {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "File: {}", self.file_path)?;
```

Comment:
To access `file_path` in `summary.rs`, The comlier think it's needed to set `file_path` public.

Reply:
Set it to `pub(crate)`

Reply:
Seems it's not bad to make it public.   
Another option it's to create a directory named`mbtiles` with two files:
* `mbtiles/mod.rs`
* `mbtiles/summary.rs`

Any help? Is it to make `file_path` public ok? I prefer the public approch.

Reply:
Sure I'll look at the whole pr soon. Need to get some sleep 😀

Reply:
It's Ok I need to handle some daily job now. See you soon.